provider/datagen/Cargo.toml

```diff
@@ -48,15 +48,18 @@ icu_segmenter = { version = "1.2.0", path = "../../components/segmenter", featur
 icu_timezone = { version = "1.2.0", path = "../../components/timezone", features = ["datagen"] }
 
 # ICU provider infrastructure
-icu_provider = { version = "1.2.0", path = "../core", features = ["std", "log_error_context", "datagen"]}
+icu_provider = { version = "1.2.0", path = "../core", features = ["std", "log_error_context", "datagen", "deserialize_json"]}
```

Comment:
Issue: Can we avoid the required deserialize_json dependency?

Reply:
`From<serde_json::Error> for DataError` is enabled by that, which would introduce a lot of boilerplate in this crate if we didn't have it.

Reply:
`deserialize_json` is one of a handful of "mutating" Cargo features in icu4x where enabling the feature is not just a no-op exposing a few more APIs; it increases the size of all BufferProvider impls. Therefore it is important that we are able to disable it. Having `icu_datagen` in your build should not force `deserialize_json` (note: Bazel and GN use global feature resolution).

Reply:
Well currently datagen enables it through `icu_provider_fs`, so this PR is not a regression in that sense. I agree it would be nice to be able to disable it, but I don't see this as blocking.